SONARJAVA-2097 fix review comments

saberduck opened the pull request:
I realized something fishy in the ProgramPoint invariant check and added FIXME to investigate it later, thus change involves bit more than what was requested

SonarTech commented:
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

SonarTech commented on java-frontend/src/main/java/org/sonar/java/se/ProgramPoint.java:
```diff
@@ -40,10 +40,13 @@ private ProgramPoint(CFG.Block block, int i) {
+    int blockSize = block.elements().size();
+    int nextPP = this.i + 1;
+    // pointing one element above block size is allowed, it represents terminator of the CFG block
+    // FIXME when RelationalSV is involved, we can point up to two elements behind, which is fishy
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1134)

m-g-sonar replied:
JIRA ticket with reproducer.

m-g-sonar replied:
So it means that what we have on master can fail?

saberduck replied:
No, condition is same as before, just rewritten for clarity. I will open the ticket and update the FIXME, I will reassign ticket back to me til it's done

m-g-sonar replied:
oh yeah, indeed, not seen the change from `<=` to `<`.

saberduck replied:
after some investigation I removed FIXME, because in fact it's valid behavior. I moved invariant check to constructor and added a comment explaining it

saberduck commented:
After some investigation I removed FIXME, because in fact it's valid behavior. I moved invariant check to constructor and added a comment explaining it
